polaris-discovery/polaris-discovery-client/src/main/java/com/tencent/polaris/discovery/client/flow/RegisterStateManager.java

```diff
@@ -55,7 +55,9 @@ public static RegisterState putRegisterState(SDKContext sdkContext,
         String registerStateKey = buildRegisterStateKey(instanceRegisterRequest);
         Map<String, RegisterState> sdkRegisterStates = REGISTER_STATES.computeIfAbsent(
                 sdkContext.getValueContext().getClientId(), clientId -> new ConcurrentHashMap<>());
-        if (sdkRegisterStates.containsKey(registerStateKey)) {
+        RegisterState existsRegisterState = sdkRegisterStates.get(registerStateKey);
+        if (existsRegisterState != null) {
+             existsRegisterState.setInstanceRegisterRequest(instanceRegisterRequest);
             return null;
```

Comment:
这里是否应该把existsRegisterState返回出去？

Reply:
> 心跳任务不需要做出变化，复用之前的心跳taskFuture即可

这里和心跳没关系，只是put的时候会返回RegisterState。如果存在已有的RegisterState，是否也要返回呢？

Reply:
/*   
* @return Return new instance register state object if it is not cached, otherwise null
  */
    public static RegisterState putRegisterState(SDKContext sdkContext,
            InstanceRegisterRequest instanceRegisterRequest) 
你是否想修改这个方法的定义，现在外部使用这个方法是按存在就返回null的逻辑开发的，现在的修改方式是存在-> 更新 -> 返回null，和方法注释的@return契合